### PR TITLE
[DON'T MERGE] acceptance tests: decommission device

### PR DESCRIPTION
**Pull request by mchalski**
Issues: MEN-1129

This is an early review, the test fails for now - needs MEN-1088 which is suspended for a couple days. 

Ready for comments however, as the implementation is finished.

@maciejmrowiec @mendersoftware/rndity

### CODE REVIEW

File: tests/tests/client.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import urllib.parse as up
 import requests
 
+
```
Comment (bboozzoo) [inline tests/tests/client.py]: hm?

File: tests/tests/test_device.py [hunk 2]
```diff
@@ -124,3 +124,40 @@ def test_get_single_device(self):
...
+        dev = Device()
+        da = DevAuthorizer()
+        # poke devauth so that device appears
+        rsp = device_auth_req(self.devapi.make_api_url("auth_requests"),
+                              da, dev)
+        assert rsp.status_code == 401
+
+        mc = SimpleManagementClient()
+        ourdev = mc.find_device_by_identity(dev.identity)
+        assert ourdev != None
```
Comment (bboozzoo) [inline tests/tests/test_device.py]: `assert outdev`

File: tests/tests/test_device.py [hunk 3]
```diff
@@ -124,3 +124,40 @@ def test_get_single_device(self):
...
+
+        # delete our device
+        # use a predetermined request id to correlate with executed workflows
+        self.delete_device(ourdev.id, {'X-MEN-RequestID':'delete_device'})
+        not_found = mc.find_device_by_identity(dev.identity)
```
Comment (bboozzoo) [inline tests/tests/test_device.py]:
```
found = ...
assert not found
```

File: tests/tests/client.py (hunk 1 above)
Comment (mchalski) [inline tests/tests/client.py]: fixed

File: tests/tests/test_device.py (hunk 2 above)
Comment (mchalski) [inline tests/tests/test_device.py]: fixed

File: tests/tests/test_device.py (hunk 3 above)
Comment (mchalski) [inline tests/tests/test_device.py]: fixed

**Comment by bboozzoo** [comment]
Acceptance tests are failing with:
```

------------------- generated xml file: /testing/results.xml -------------------
=================================== FAILURES ===================================
[31m[1m________________________ TestDevice.test_delete_device _________________________[0m
testing/tests/test_device.py:160: in test_delete_device
[1m    assert res['totalHits'] > 0[0m
[31m[1mE   assert 0 > 0[0m
[31m[1m====================== 1 failed, 7 passed in 6.71 seconds ======================[0m
Stopping acceptancetests_mender-device-adm_1 ... 
```

**Comment by bboozzoo** [comment]
waiting for https://tracker.mender.io/browse/MEN-1088

**Comment by maciejmrowiec** [comment]
lets aim to merge this before stable (will need 1.1.x cherry-pick as well )

**Comment by bboozzoo** [comment]
there are merge conflicts, @mchalski will need to fix it first

**Comment by mchalski** [comment]
@maciejmrowiec @bboozzoo rebased & merged

**Pull request merged**